assets/training/distillation/src/validate_pipeline.py

```diff
@@ -186,7 +186,7 @@ def _validate_inference_parameters(self):
 
     def _validate_number_of_records(self, size: int):
         """Validate number of records in the dataset."""
-        if size < MIN_RECORDS_FOR_FT:
+        if size < MIN_RECORDS_FOR_FT and self._args.data_generation_task_type != DataGenerationTaskType.MATH:
```

Comment:
Could you please share reasoning behind this check? AFAIU, even math would need to follow min row req.

Reply:
I used 3 math datasets to test this. Some math datasets don't have a validation set so it is taken from the training data. The data science team had a 90-10 split for when no validation set was available. For one dataset, the Multi-Arith dataset, the number of entries for validation following the 90-10 split was 42 records. I was able to finetune the llama-3.1-8b model successfully and there were improvements in the model's accuracy as well even with this small amount of validation data.